astropy/modeling/parameters.py

```diff
@@ -728,22 +728,22 @@ def __rtruediv__(self, val):
         return val / self.value
 
     def __eq__(self, val):
-        return (np.asarray(self) == np.asarray(val)).all()
+        return np.asanyarray(self) == np.asanyarray(val)
```

Comment:
Note that `asanyarray` will just call the `__array__` method and thus return a plain array! You need `np.asanyarray` in `__array__` for this to work.

With that, I think it this should be something like
```
def __eq__(self, val):
    return self.__array__().__eq__(val)
```
This would let `ndarray.__eq__` take care of interpreting `val` (no need to do it yourself).

The above assumes you want `Parameter` to always look array-like (probably good). But maybe `_value` is already like that? If so, then maybe even cleaner would be:
```
    if self._model:
        return self.value.__eq__(val)
    else:
        return False
```
For the comparisons, you are allowed to raise an error, so it would just be
```
def __lt__(self, val):
    return self.value.__lt__(val)
```